File: 신창호/Lv2/숫자의 표현.java
class Solution {
    public int solution(int n) {
        //  n - 1 / 2  나머지가 없으면 가능
        //  n / 3
        //  n -2  / 4
        //  n / 5
        // n -3 / 6  0, 1, 2,3,4,5
        //
        // n - 5/10    -3, -2, -1, 0, 1,2,3 , 4,5,6
        int answer = 0;
        for(int i = 1; i <= n; i++){
            if(i%2 == 0){ // 짝수
                if((n - (i/2)) % i == 0 && ((n - (i/2)) / i) >= i/2 ){
                    answer++;
                }
            }
            else{// 홀수
                if(n%i == 0 &&  n/i > i/2){
                    answer++;
                }
            }
        }


        return answer;

    }
}

File: 조은정/Lv.2/숫자의 표현.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(int n)
{
    int answer = 0;
    long long md, st, en, sum;
    for (int i = 1; i <= n; ++i)
    {
        md = n / i;
        if ((i & 1) && md <= i / 2)
            break;
        if (!(i & 1) && md < i / 2)
            break;
        if (i & 1)
        { // 홀수개
            sum = i * md;
            if (sum == n)
                ++answer;
        }
        else
        { // 짝수개
            sum = (i - 1) * md;
            if (sum + md + i / 2 == n)
                ++answer;
            if (sum + md - i / 2 == n)
                ++answer;
        }
    }
    return answer;
}